### PR TITLE
MINOR: Update JavaDoc for KTable helper interfaces

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
@guozhangwang @dguy @enothereska @miguno

**Review by dguy — APPROVED** [review]
LGTM

File: streams/src/main/java/org/apache/kafka/streams/kstream/Aggregator.java [hunk 1]
```diff
@@ -17,22 +17,36 @@
...
+ * @see Initializer
+ * @see KGroupedStream#aggregate(Initializer, Aggregator, org.apache.kafka.common.serialization.Serde, String)
+ * @see KGroupedStream#aggregate(Initializer, Aggregator, org.apache.kafka.streams.processor.StateStoreSupplier)
+ * @see KGroupedStream#aggregate(Initializer, Aggregator, Windows, org.apache.kafka.common.serialization.Serde, String)
+ * @see KGroupedStream#aggregate(Initializer, Aggregator, Windows, org.apache.kafka.streams.processor.StateStoreSupplier)
+ * @see Reducer
  */
-public interface Aggregator<K, V, T> {
+@InterfaceStability.Unstable
+public interface Aggregator<K, V, VA> {
```
Comment (enothereska) [inline streams/src/main/java/org/apache/kafka/streams/kstream/Aggregator.java:41]: Why are two letters better than one? Just curious.

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/kstream/Aggregator.java:41]: It's not about the number of letters. But what does `T` stand for? "Type" is not very meaningful. `VA` is for "Value Aggregate". (ie, `K` -> "key type" -- `V` -> "value type" -- "value aggregate type"). This is more meaningful -- in some other classes we also use stuff like `K`, `K1` -- this in no helpful at all to the user -- it almost like using `A`, `B`.

File: streams/src/main/java/org/apache/kafka/streams/kstream/Reducer.java [hunk 2]
```diff
@@ -17,19 +17,33 @@
 
 package org.apache.kafka.streams.kstream;
 
+import org.apache.kafka.common.annotation.InterfaceStability;
+
 /**
  * The {@link Reducer} interface for combining two values of the same type into a new value.
+ * In contrast to {@link Aggregator} the result type must be the same as the input type.
+ * <p>
+ * The provided values can be either original values from input {@code <key,value>} pair records or be a previously
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/kstream/Reducer.java]: There is a `KeyValue` class that we can reference here.

**Review by guozhangwang — COMMENTED** [review]
One minor comment, otherwise LGTM.

**Comment by mjsax** [comment]
@guozhangwang Updated

**Comment by guozhangwang** [comment]
Thanks, merged to trunk.

**Pull request closed without merging**